### PR TITLE
Migrator V2

### PR DESCRIPTION
**Pull request by afkbyte**
Migrator V2 complete with the following features:

- Approver and Blacklister roles
- Blacklist functionality
- migration request approval over user migration

### CODE REVIEW

File: contracts/Migration/MantleTokenMigratorV2.sol [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+    address public owner;
+
+    /// @dev A mapping of addresses with the ability to approve token migrations
+    mapping(address => bool) public approvers;
+
+    /// @dev A mapping of addresses with the ability blacklist addresses from migration
+    mapping(address => bool) public blacklisters;
+
+    /// @dev A mapping of addresses blacklisted from migration
+    mapping(address => address) public blacklistedAddresses;
```
Comment (Tri-stone) [inline contracts/Migration/MantleTokenMigratorV2.sol:36]: use `mapping(address => bool) public` maybe better and more cost-economical

Comment (afkbyte) [reply contracts/Migration/MantleTokenMigratorV2.sol:36]: lol good catch i don't know why i did that

**Review by Tri-stone — APPROVED** [review]
LGTM

File: contracts/Migration/MantleTokenMigratorV2.sol [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+
+    /// @notice Blacklists an address from token migration
+    /// @dev emits an {AddressBlacklisted} event
+    /// @dev Requirements:
+    ///     - The caller must have the Blacklister role
+    /// @param _address The address to blacklist
+    function blacklistAddress(address _address) public onlyBlacklister {
+        blacklistedAddresses[_address] = true;
+
+        emit AddressBlacklisted(msg.sender, _address);
```
Comment (aodhgan) [inline contracts/Migration/MantleTokenMigratorV2.sol:322]: we could recycle and use something like `BlacklistModified(address, address, bool)` to reduce the number of events (and indexer handlers) but strictly a suggestion.

File: contracts/Migration/MantleTokenMigratorV2.sol [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+            revert MantleTokenMigrator_InvalidFundingToken(_tokenAddress);
+        }
+
+        // we can only defund BIT or MNT into the predefined treasury address
+        ERC20(_tokenAddress).safeTransfer(treasury, _amount);
+
+        emit ContractDefunded(treasury, _tokenAddress, _amount);
+    }
+
+    /// @notice Sweeps a specified amount of tokens to an arbitrary address
```
Comment (aodhgan) [inline contracts/Migration/MantleTokenMigratorV2.sol:391]: suggestion: maybe add a note here to say this is for accidental transfers

File: contracts/Migration/MantleTokenMigratorV2.sol (hunk 2 above)
Comment (afkbyte) [reply contracts/Migration/MantleTokenMigratorV2.sol]: combined

File: contracts/Migration/MantleTokenMigratorV2.sol (hunk 3 above)
Comment (afkbyte) [reply contracts/Migration/MantleTokenMigratorV2.sol]: done

**Pull request merged**